Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -194,12 +196,88 @@ void AODProducerWorkflowDPL::addToMFTTracksTable(mftTracksCursorType& mftTracksC
+                                                         TracksExtraCursorType& tracksExtraCursor,
+                                                         mftTracksCursorType& mftTracksCursor)
+{
+  const auto& tpcClusRefs = data.getTPCTracksClusterRefs();
+  const auto& tpcClusShMap = data.clusterShMapTPC;
+  const auto& tpcClusAcc = data.getTPCClusters();
+
+  for (int src = GIndex::NSources; src--;) {
+    int start = trackRef.getFirstEntryOfSource(src);
+    int end = start + trackRef.getEntriesOfSource(src);
+    LOG(DEBUG) << "Unassigned tracks: src = " << src << ", start = " << start << ", end = " << end;
+    for (int ti = start; ti < end; ti++) {
+      TrackExtraInfo extraInfoHolder;
+      auto& trackIndex = GIndices[ti];
+      if (GIndex::includesSource(src, mInputSources)) {
+        if (src == GIndex::Source::MFT) { // MFT tracks are treated separately since they are stored in a different table
+          const auto& track = data.getMFTTrack(trackIndex.getIndex());
+          addToMFTTracksTable(mftTracksCursor, track, collisionID);
+        } else {
+          auto contributorsGID = data.getSingleDetectorRefs(trackIndex);
+          const auto& trackPar = data.getTrackParam(trackIndex);
+          if (contributorsGID[GIndex::Source::ITS].isIndexSet()) {
+            const auto& itsOrig = data.getITSTrack(contributorsGID[GIndex::ITS]);
+            extraInfoHolder.itsClusterMap = itsOrig.getPattern();
+          }
+          if (contributorsGID[GIndex::Source::TPC].isIndexSet()) {
+            const auto& tpcOrig = data.getTPCTrack(contributorsGID[GIndex::TPC]);
+            extraInfoHolder.tpcInnerParam = tpcOrig.getP();
+            extraInfoHolder.tpcChi2NCl = tpcOrig.getNClusters() ? tpcOrig.getChi2() / tpcOrig.getNClusters() : 0;
+            extraInfoHolder.tpcSignal = tpcOrig.getdEdx().dEdxTotTPC;
+            uint8_t shared, found, crossed; // fixme: need to switch from these placeholders to something more reasonable
+            countTPCClusters(tpcOrig, tpcClusRefs, tpcClusShMap, tpcClusAcc, shared, found, crossed);
+            extraInfoHolder.tpcNClsFindable = tpcOrig.getNClusters();
+            extraInfoHolder.tpcNClsFindableMinusFound = tpcOrig.getNClusters() - found;
+            extraInfoHolder.tpcNClsFindableMinusCrossedRows = tpcOrig.getNClusters() - crossed;
+            extraInfoHolder.tpcNClsShared = shared;
+          }
+          if (contributorsGID[GIndex::Source::ITSTPCTOF].isIndexSet()) {
+            const auto& tofMatch = data.getTOFMatch(contributorsGID[GIndex::Source::ITSTPCTOF]);
+            extraInfoHolder.tofChi2 = tofMatch.getChi2();
+            const auto& tofInt = tofMatch.getLTIntegralOut();
+            float intLen = tofInt.getL();
+            extraInfoHolder.length = intLen;
+            float mass = o2::constants::physics::MassPionCharged; // default pid = pion
+            float expSig = tofMatch.getSignal();
+            extraInfoHolder.tofSignal = expSig;
```

Comment:
I see a problem here: in the Run2 -> Run3 AOD conversion we will a truncated float value, 
https://github.com/alisw/AliPhysics/blob/bb0e58318cd9d7fc90d659a724643b0f2b1c623e/RUN3/AliAnalysisTaskAO2Dconverter.cxx#L1650
to https://github.com/alisw/AliPhysics/blob/bb0e58318cd9d7fc90d659a724643b0f2b1c623e/RUN3/AliAnalysisTaskAO2Dconverter.h#L326.
This should be fine for Run2 AliESDtrack, where the time is defined wrt to the trigger, but, as @noferini explained, in Run3, with the time defined wrt the TF start, we should use double. And, we cannot redefine this time wrt some other reference (e.g. interaction time) since in general the track is not guaranteed to be related to any reference.
I would propose to change `float expSig` and `extraInfoHolder.tofSignal` to double and add it on
https://github.com/AliceO2Group/AliceO2/blob/af43e338c5ad94efa896f9afeb38231bdd98f085/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L175 directly, w/o truncation.
Then, the `AliAnalysisTaskAO2Dconverter` should be fixed in AliPhysics, and the lines
https://github.com/AliceO2Group/AliceO2/blob/2360d52d0484a1880a77e519cdfc6705d04394a3/Framework/Core/include/Framework/AnalysisDataModel.h#L209 and https://github.com/AliceO2Group/AliceO2/blob/2360d52d0484a1880a77e519cdfc6705d04394a3/Framework/Core/include/Framework/AnalysisDataModel.h#L214-L215 should be redefined with `double tofSignal`.
Pinging @pzhristov and @jgrosseo

Reply:
I have a comment regarding propagation of tof time to aod: float or double?
Since tof time is filled in a method called with the collisionID as argument I would expect that at this stage you already know at least what is the BC candidate. In such a case you should be able to subtract a collision time with a precision depending on what you have. I think it doesn't make sense propagate to aod tof time referred to the the TF (even if I didn't check the logic of AOD producer).

Reply:
@noferini see my comment below: for unassigned tracks the collisionID and `interactionTime` are dummy parameters.
But even for assigned tracks the current `interactionTime` is not guaranteed to be even BC-precise (and, btw, it is rounded to ns): https://github.com/AliceO2Group/AliceO2/blob/af43e338c5ad94efa896f9afeb38231bdd98f085/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L762-L767

Reply:
I do not think this fully solves it.

For assigned tracks, we could store wrt to track.collision().bc()

But as you point out unassigned tracks, do not have a collision or bc. So storing with respect of TF start, also does not work, because we do not store the TF start. As Ruben pointed out earlier, we can calculate TF start from any BC in a TF. However, we merge TFs into DFs, and therefore you have to select the *right* BC for the TF start calculation.

In principle, each unassigned track appears in the AmbiguousTrack table where it is assigned a BC slice or a collision array. One could define the time wrt the first BC in that table, but we have to look up the corresponding association track->ambiguoustrack while the index goes the other way round...